app/controllers/chatroom_controller.rb

```diff
@@ -0,0 +1,45 @@
+class ChatroomController < ApplicationController    
+   before_action :load_chat_rooms, only: [:show ]
+
+    def index
+        @chatrooms = Chatroom.all
+    end
+
+    def new
+        @chatroom = Chatroom.new
+    end
+
+    def create
+        @chatroom = Chatroom.create(load_params)
+        
+        if @chatroom.save               
+            redirect_to root_path
+
+        else
+            render :new
+        end 
+    end
+
+    def edit
+    
+    end
+
+    def show 
+        #byebug
+        @chatroom_message = Message.new chatroom: @chatroom 
+        
+        @chatroom_messages = @chatroom.messages.includes(:user)
+    end
```

Comment:
This controller should have the `new` action and that is for the `Message` model.

Reply:
Remove your other codes for a while. I'll let you do some addons when we're done on our chat app MVP